pinot-spi/pom.xml

```diff
@@ -36,6 +36,82 @@
 
   <build>
     <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <version>3.7.1</version>
+        <executions>
+          <execution>
+            <id>copy-pinot-plugins</id>
+            <!-- The org.apache.pinot.spi.plugin.PluginManagerTest has tests that requires plugin jars.
+                 These jars could have been added to src/test/resources/plugins, but that would increase the size of
+                 source repository a lot. Instead, let Maven download these jars and put them at the expected location.
+            -->
+            <phase>generate-test-resources</phase>
+            <goals>
+              <goal>copy</goal>
+            </goals>
+            <configuration>
+              <artifactItems>
+                <artifactItem>
+                  <groupId>org.apache.pinot</groupId>
+                  <artifactId>pinot-dropwizard</artifactId>
+                  <version>0.10.0</version>
```

Comment:
> Can we avoid hard code versions for these libraries? A

We need to use an old version instead of the latest one because what we are doing here is to copy the version the repository (ideally the local one, but will fetch from the remote if it is not there). If we try to fetch the version we are compiling that will fail (because this is executed before our build of the plugin is installed).

>  Also, why is the version 0.10.0?

Any version is valid. Remember we only use this in a test to verify that realms are working as expected (ie that X class is read from the plugin while Y is read from pinot-framework).

Reply:
We need a stable jar from which we exactly know which class are there and which one aren't.  That's why we cannot and shound not use the latest snapshot here. Suppose we did that, there could be a situation where a change to pinot-dropwizard could make a test of pinot-spi fail. That is really unexpected!

Reply:
I see. I missed the fact that this part is used to generate resources for the test